oauth2_http/javatests/com/google/auth/oauth2/MockMetadataServerTransport.java

```diff
@@ -83,134 +85,152 @@ public void setIdToken(String idToken) {
   @Override
   public LowLevelHttpRequest buildRequest(String method, String url) throws IOException {
     if (url.equals(ComputeEngineCredentials.getTokenServerEncodedUrl())) {
+      return getMockRequestForTokenEndpoint(url);
```

Comment:
This diff might be tricky to read. What I did is I factored out all the if-statements contents in a separate private methods. Some are big statements that made this method hard to read.